mentat/commands.py

```diff
@@ -338,3 +341,48 @@ def argument_names(cls) -> list[str]:
     @classmethod
     def help_message(cls) -> str:
         return "Shows all files currently in Mentat's context"
+
+
+class ConfigCommand(Command, command_name="config"):
+    async def apply(self, *args: str) -> None:
+        session_context = SESSION_CONTEXT.get()
+        stream = session_context.stream
+        config = session_context.config
+        if len(args) == 0:
+            stream.send("No config option specified", color="yellow")
+        else:
+            setting = args[0]
+            if hasattr(config, setting):
+                if len(args) == 1:
+                    value = getattr(config, setting)
+                    description = attr.fields_dict(Config)[setting].metadata.get(
+                        "description"
+                    )
+                    stream.send(f"{setting}: {value}")
+                    if description:
+                        stream.send(f"Description: {description}")
+                elif len(args) == 2:
+                    value = args[1]
+                    if attr.fields_dict(Config)[setting].metadata.get(
+                        "no_midsession_change"
+                    ):
+                        stream.send(
+                            f"Cannot change {setting} mid-session. Please restart"
+                            " Mentat to change this setting.",
+                            color="yellow",
+                        )
+                        return
+                    setattr(config, setting, value)
```

Comment:
Could this set it to a type it shouldn't be? I also thing the Config should have functions for getting and setting config attributes instead of using setattr and getattr

Reply:
I should read the docs for attrs more carefully but I believe this is safe and will use the conversions I specified or the obvious ones e.g. if you try `/config temperature lol` you get the error message: `Error processing command arguments: could not convert string to float: 'lol'`

Reply:
It'd probably be a good idea to add validators though because you can set temperature to a number outside 0 to 1.

Reply:
I added validators for format, temperature, maximum_context and auto_context.

Reply:
I'm pretty sure setattr and getattr have nothing to do with attrs; they're just built into python. Either way, looks good to me!

Reply:
Yeah `setattr` and `getattr` are built in but somehow when you use them on a class with `attr` attributes `attr` is able to run its converts and validaters and either change what you set it to (like automatically calling int) or throw an error. I should probably read the attrs source to learn how it works.